helix-term/src/ui/statusline.rs

```diff
@@ -180,7 +180,13 @@ where
         ))
     } else {
         // If not focused, explicitly leave an empty space instead of returning None.
-        Cow::Borrowed("     ")
+        let mode_str = match context.editor.mode() {
+            Mode::Insert => &modenames.insert,
+            Mode::Select => &modenames.select,
+            Mode::Normal => &modenames.normal,
+        };
+        let width = mode_str.chars().count() + 2;
```

Comment:
Here we should use `width` (by importing `helix_core::unicode::width::UnicodeWidthStr`) rather than counting chars. Counting chars doesn't work properly if these strings are configured to be grapheme clusters like emoji